front/src/types/safeCast.ts

```diff
@@ -0,0 +1,19 @@
+import { Order, SortOptions, TimeOption } from "./logDisplayOptions";
+
+export const safeCastToTimeOption = (value: string | null): (TimeOption | undefined) => {
+    if (["Last hour", "Last day", "Last week"].includes(value ?? ""))
+        return value as TimeOption;
+    return undefined;
```

Comment:
(refactor) I think errors are usually thrown if it is not that type, instead of returning undefined

Reply:
oh sorry, misunderstood, this can be undefineable?